### PR TITLE
Feat/chart lint release

### PR DESCRIPTION
**Pull request by paologallinaharbur**
Adding chart testing and linting

Fix https://github.com/newrelic/newrelic-infra-operator/issues/143

### CODE REVIEW

File: charts/newrelic-infra-operator/Chart.yaml
Comment (kang-makes) [inline charts/newrelic-infra-operator/Chart.yaml:5]: Until the `0.6.0` version is deployed an orphan chart is going to be available.

It is a NIT that does not block the release because we are not using `nri-bundle` yet but is something we have to care in the future ;)

File: .github/workflows/ci.yaml
Comment (kang-makes) [inline .github/workflows/ci.yaml:69]: I know that I am saying this like 3 PRs before we were merging this pattern but I thought about this right now...

WDYT if we have the upgradability test separated in a separate job? In case we have to change the configuration it could fail and we can ignore it. Or even add a tag like `breaking-chart` and don't run the chart at all.

I would not like you to tackle this right now, just thinking out loud.

**Review by kang-makes — APPROVED** [review]
LGTM. This is still not in production so I don't care about the release without image.

File: charts/newrelic-infra-operator/Chart.yaml
Comment (paologallinaharbur) [reply charts/newrelic-infra-operator/Chart.yaml:5]: you mean [this](https://github.com/newrelic/newrelic-infra-operator/releases/tag/v0.6.0) one? Or do you mean something else?

File: .github/workflows/ci.yaml
Comment (paologallinaharbur) [reply .github/workflows/ci.yaml:69]: Yes whenever we change the config the upgrade process could fail. I agree that is not ideal, but testing the upgrade process is important. 

Since it will rarely happen to have a configuration breaking change we can test it manually and ignore the job output

File: charts/newrelic-infra-operator/Chart.yaml
Comment (kang-makes) [reply charts/newrelic-infra-operator/Chart.yaml:5]: I thought you were going to release it again because of this change: https://github.com/newrelic/newrelic-infra-operator/pull/157/files#diff-8df2af49373ef833809e6615c110f9c7ac8fdacf8dc6f6c0215ba948e6666046R62-R64

What is that spurious change?

Comment (paologallinaharbur) [reply charts/newrelic-infra-operator/Chart.yaml:5]: that is a change to make a flaky test less flaky

**Pull request merged**
